### PR TITLE
Fixed several bugs related to infowindows and tooltips

### PR DESCRIPTION
**Pull request by alonsogarciapablo**
Fixes https://github.com/CartoDB/cartodb.js/issues/1586, https://github.com/CartoDB/cartodb.js/issues/1585, and https://github.com/CartoDB/cartodb.js/issues/1582.

@xavijam 🙏

### CODE REVIEW

File: src/geo/map/cartodb-layer.js [hunk 1]
```diff
@@ -36,6 +36,7 @@ var CartoDBLayer = LayerModelBase.extend({
     this.unset('legends');
 
     this.bind('change', this._onAttributeChanged, this);
+    this.infowindow.fields.bind('reset', this._reloadVis, this);
```
Comment (xavijam) [inline src/geo/map/cartodb-layer.js]: `reset`, `add` and `remove` or only needed `reset`?

File: src/vis/vis.js [hunk 2]
```diff
@@ -315,8 +315,21 @@ var VisModel = Backbone.Model.extend({
...
+    var successCallback = options.success;
+    var errorCallback = options.error;
+
+    options = _.extend({
+      includeFilters: true,
+      success: function () {
+        this.trigger('reloaded');
+        successCallback && successCallback();
+      }.bind(this),
+      error: function () {
```
Comment (xavijam) [inline src/vis/vis.js]: Don't we want to return the error in the `errorCallback`?

File: src/geo/map/cartodb-layer.js (hunk 1 above)
Comment (alonsogarciapablo) [inline src/geo/map/cartodb-layer.js]: Fields are only reset at the moment, but I'll add `add` and `remove` too 👍

File: src/vis/vis.js (hunk 2 above)
Comment (alonsogarciapablo) [inline src/vis/vis.js]: We're not currently receiving any error here, so there's no error to "return"...

**Pull request merged**